search-improvements.rst

```diff
@@ -0,0 +1,143 @@
+.. post:: August 14, 2018
+   :tags: gsoc, search, feature
+   :author: Safwan
+   :location: DHA
+
+Improved Search
+==================================
+Have you ever used the search functionality inside documentations hosted by Read The Docs?
+Mostly yes. Search can always be improved and in the era of Search Engines, the
+documentation search also plays a big role for the users. We have realized the importance,
+and got our GSoC student interested to take the challenge. The main goal was to refactor the
+search code together with upgrading the backend Search engine, as well as adding more features
+to the search functionality like exact match search, case insensitive search,
+search as you type, suggestions etc.
+
+Background
+^^^^^^^^^^
+Search is a vital part of any documentation hosting platform as people can get the
+information they need. As a documentation hosting platform, same rule applies for
+Read The Docs. Because of having a small core team, the search functionality
+of Read The Docs has lagged behind for quite a while now. Initially the search code
+was voluntarily contributed by `Rob Hudson`_,  `back in 2013`_ and then improved by other
+contributors. The time span and continuous technology growing made the search
+module got older and it had become outdated. Moreover The search infrastructure was already
+vulnerable as we were using `Elasticsearch 1.3.x`_ which was already reached its
+End of Life in 2016. Therefore, the upgradation of search infrastructure was badly needed.
+
+Google Summer of Code
+~~~~~~~~~~~~~~~~~~~~~
+Google Summer of Code is a global program where students work with an open source organization
+on a 3 month programming project. The core team of Read The Docs proposed some `Project Ideas`_,
+one of them was **Refactor & improve our search code**. Our contributor (me) `Safwan Rahman`_,
+an undergraduate university student and open source contributor from Bangladesh was keen
+to get his hand dirty with Elasticsearch_ and grasped the opportunity to do so by applying
+for this project and got accepted.
+
+Safwan_ has worked full time for last 3 months to upgrade the whole codebase
```

Comment:
I'd use bullets for this section. It will come across better than a big block of text links.